### PR TITLE
Update Offset mapping

### LINKED ISSUE #880: offset_mapping 偏移错误【错误原因2】
错误原因：`where`跟`is`中间有2个空格，空格导致的偏移，使用basic_tokenizer后把空格都去掉了。
# 
```python
from paddlenlp.transformers import BertTokenizer
tokenizer = BertTokenizer.from_pretrained("bert-base-uncased")

examples = [{'id': '5737a5931c456719005744e9',
 'title': 'Force',
 'context': 'where  is the mass of the object,  is the velocity of the object and  is the distance to the center of the circular path and  is the unit vector pointing in the radial direction outwards from the center. This means that the unbalanced centripetal force felt by any object is always directed toward the center of the curving path. Such forces act perpendicular to the velocity vector associated with the motion of an object, and therefore do not change the speed of the object (magnitude of the velocity), but only the direction of the velocity vector. The unbalanced force that accelerates an object can be resolved into a component that is perpendicular to the path, and one that is tangential to the path. This yields both the tangential force, which accelerates the object by either slowing it down or speeding it up, and the radial (centripetal) force, which changes its direction.',
 'question': 'What force changes an objects direction of travel?',
 'answers': ['centripetal',
  'unbalanced centripetal force',
  'unbalanced centripetal force',
  'centripetal'],
 'answer_starts': [837, 224, 224, 837],
 'is_impossible': False}]
def prepare_validation_features(examples, tokenizer):
    contexts = [examples[i]['context'] for i in range(len(examples))]
    questions = [examples[i]['question'] for i in range(len(examples))]

    tokenized_examples = tokenizer(
        questions,
        contexts,
        stride=128,
        max_seq_len=512)

    # For validation, there is no need to compute start and end positions
    for i, tokenized_example in enumerate(tokenized_examples):
        # Grab the sequence corresponding to that example (to know what is the context and what is the question).
        sequence_ids = tokenized_example['token_type_ids']

        # One example can give several spans, this is the index of the example containing this span of text.
        sample_index = tokenized_example['overflow_to_sample']
        tokenized_examples[i]["example_id"] = examples[sample_index]['id']

        # Set to None the offset_mapping that are not part of the context so it's easy to determine if a token
        # position is part of the context or not.
        tokenized_examples[i]["offset_mapping"] = [
            (o if sequence_ids[k] == 1 else None)
            for k, o in enumerate(tokenized_example["offset_mapping"])
        ]

    return tokenized_examples[0]
outputs = prepare_validation_features(examples,tokenizer)
for ids,offsetmap in zip(outputs["input_ids"],outputs["offset_mapping"]):
    if offsetmap:
        print(offsetmap,examples[0]["context"][offsetmap[0]:offsetmap[1]])
        print("="*20)
```
# 错误结果
```
(0, 5) where
====================
(6, 8)  i
====================
(9, 12)  th
====================
(13, 17)  mas
====================
(18, 20)  o
====================
(21, 24)  th
====================
(25, 31)  objec
====================
(31, 32) t
====================
(33, 35)   
====================
(36, 39) s t
====================
(40, 48) e veloci
====================
(49, 51) y 
====================
(52, 55) f t
====================
(56, 62) e obje
====================
(63, 66) t a
```

### PR DESCRIPTION
**Pull request by JunnYu**
更新offset mapping计算的方法，主要参考了https://github.com/bojone/bert4keras/blob/master/bert4keras/tokenizers.py#L372。

### CODE REVIEW

**Comment by yingyibiao** [comment]
mark

File: paddlenlp/transformers/tokenizer_utils.py [hunk 1]
```diff
@@ -1390,6 +1344,47 @@ def get_input_ids(text):
 
         return batch_encode_inputs
 
+    def rematch(self, text):
```
Comment (smallv0221) [inline paddlenlp/transformers/tokenizer_utils.py:1347]: Is the name get_offset_mapping better?

File: paddlenlp/transformers/tokenizer_utils.py [hunk 2]
```diff
@@ -1390,6 +1344,47 @@ def get_input_ids(text):
...
+        """
+            changed from https://github.com/bojone/bert4keras/blob/master/bert4keras/tokenizers.py#L372
```
Comment (smallv0221) [inline paddlenlp/transformers/tokenizer_utils.py:1349]: Modified from https://github.com/bojone/bert4keras/blob/master/bert4keras/tokenizers.py#L372

**Review by smallv0221 — CHANGES_REQUESTED** [review]
Excellent work! Only a few small problems.

**Review by smallv0221 — APPROVED** [review]

**Comment by smallv0221** [comment]
LGTM! Thanks for your contribution.

**Pull request merged**